amqp/src/test/scala/akka/stream/alpakka/amqp/AmqpProxyConnection.scala

```diff
@@ -0,0 +1,68 @@
+package akka.stream.alpakka.amqp
+
+import java.net.InetAddress
+import java.util
+
+import com.rabbitmq.client._
+
+class AmqpProxyConnection(delegate: Connection) extends Connection {
```

Comment:
A dumb, easily extensible proxy. I'd rather use `Mockito.spy` though I had issues with it as mentioned [here](https://github.com/akka/alpakka/pull/887#issuecomment-380319279)